plugins/mcreator-localization/lang/texts.properties

```
@@ -1207,13 +1208,24 @@ blockly.block.set_event_result.line2=Works only with global triggers with a resu
 blockly.block.call_procedure=Call procedure
 blockly.block.procedure_retval=Call procedure and get return value
 blockly.block.call_procedure.at=\ at
+blockly.block.block_predicate_inside_world=Position offset by x: %1 y: %2 z: %3 is inside world bounds
+blockly.block.block_predicate_has_sturdy_face=Block at position offset by x: %1 y: %2 z: %3 has solid face
+blockly.block.block_predicate_is_air=Is air at current position
+blockly.block.block_predicate_is_air_or_water=Is air or water at current position
+blockly.block.block_predicate_not=Not %1
+blockly.block.block_predicate_replaceable=Block at position offset by x: %1 y: %2 z: %3 is of replaceable material
+blockly.block.block_predicate_solid=Block at position offset by x: %1 y: %2 z: %3 is of solid material
+blockly.block.block_predicate_would_survive=%1 would survive at position offset by x: %2 y: %3 z: %4
 blockly.block.feature_block_pile=Block pile feature with block: %1
 blockly.block.feature_huge_brown_mushroom=Huge brown mushroom feature with cap: %1 stem: %2 radius: %3
 blockly.block.feature_huge_red_mushroom=Huge red mushroom feature with cap: %1 stem: %2 radius: %3
 blockly.block.feature_lake=Lake feature with fluid: %1 border: %2
 blockly.block.feature_simple_block=Simple block feature with block: %1
 blockly.block.placement_biome_filter=Only if biome at current position has this feature
+blockly.block.placement_block_predicate_filter=Only if condition %1 is met
 blockly.block.placement_block_survival_filter=Only if %1 would survive at current position
+blockly.block.placement_environment_scan=Scan at most %1 blocks going %2 until condition %3 is met
+blockly.block.placement_environment_scan_advanced=Scan at most %1 blocks going %2 while condition %4 is verified, until condition %3 is met
```

I'd better use the **"passes"** word.
```suggestion
blockly.block.placement_environment_scan_advanced=Scan at most %1 blocks going %2 while condition %4 passes, until condition %3 is met
```

---

Not sure if the suggestion defines what this block does though.

But it is still not clear what condition %4 does to me. %3 controls when the scan is complete, but what does %4 control?

---

It's a condition that must pass at every step. If the condition is false, the scan is aborted

---

Ok so if %4 fails, it will stop the scan and not place the feature, unless %3 passes before %4 fails or the number of blocks exceeds, then the feature is placed?

---

If this is the case, then maybe the suggestion of Defeatomizer is ok, so _passes_ or maybe _is true_